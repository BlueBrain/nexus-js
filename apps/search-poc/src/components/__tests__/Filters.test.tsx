import * as renderer from "react-test-renderer";
import Filter from '../Filters';
import React from "react";
import { shallow, mount } from "enzyme";

it('filter renders correctly', () => {
    const props = { appliedFilters :  {}, filters: [{name: 'filterName', values:[{ id: 'filterId', label: 'filterLabel'}] }], updateFilters: jest.fn};
    const result = shallow(<Filter {...props} />);
    expect(result).toMatchSnapshot();
});

