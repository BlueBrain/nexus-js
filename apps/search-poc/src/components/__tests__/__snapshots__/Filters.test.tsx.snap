// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter renders correctly 1`] = `
<div
  className="filter-list"
>
  <Collapse
    bordered={false}
    expandIconPosition="left"
    openAnimation={
      Object {
        "appear": [Function],
        "enter": [Function],
        "leave": [Function],
      }
    }
  >
    <CollapsePanel
      header={
        <div
          style={
            Object {
              "display": "flex",
              "justifyContent": "space-between",
            }
          }
        >
          <div
            style={
              Object {
                "width": "60%",
              }
            }
          >
            filterName
          </div>
          <div
            style={
              Object {
                "width": "20%",
              }
            }
          >
            <Badge
              count={0}
              dot={false}
              overflowCount={99}
              showZero={false}
              style={
                Object {
                  "backgroundColor": "#44c7f4",
                }
              }
            />
          </div>
          <div>
            1
          </div>
        </div>
      }
      key="$0"
    >
      <CheckboxGroup
        onChange={[Function]}
        options={Array []}
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <ul
          className="checkbox-list"
        >
          <li
            key="filterId"
          >
            <Checkbox
              indeterminate={false}
              value="filterId"
            >
              filterLabel
            </Checkbox>
          </li>
        </ul>
      </CheckboxGroup>
    </CollapsePanel>
  </Collapse>
</div>
`;
